### ISSUE TITLE
macOS Installation nitpicks and a crash

### ISSUE THREAD

**Issue by noahkuhn** [DidierRLopes/GamestonkTerminal#6]
Very impressive work, thanks for offering this up to the community.

Installation took a few passes. In the end, I needed to do these installations after installing the requirements.txt

pip uninstall pandas_ta 
pip install pandas_ta
pip install python-Levenshtein
pip install lunarcalendar
pip install fbprophet

Not sure why I had to uninstall pandas, but doing so let me continue.

So far, running disc > sectors gives a segfault. I have added my AlphaVantage key although to be fair I just signed up for this project so I'm not positive it's active?

**Comment by sheckylovejoy** [comment]
Same here. python-Levenshtein and fbprophet were the issues for me

**Comment by noahkuhn** [comment]
Seg fault happens on pretty much everything when using python 3.8.7, going to try and use 3.6.8 like the author.

**Comment by sheckylovejoy** [comment]
Yeah same

**Comment by sheckylovejoy** [comment]
So some other stuff on Mac
Python 3.92; Mac OS 11.2.1

On all the Reddit functions, I got this error
`'Version 7.1.4 of praw is outdated. Version 7.2.0 was released 1 day ago.
`
pip uninstall praw
pip install praw

No error now, but none of the Reddits work.


Also in the sentiment function it churns away, loads ~/.flair/models/sentiment-en-mix-distillbert_3.1.pt and exits

Thanks for the work. Very nice

**Comment by noahkuhn** [comment]
Spent an hour banging my head against a wall to get 3.6.8 installed on macOS 11.2.1, and now that I've finally got it, it's still segfaulting. Not sure what to try at this point.

**Comment by yonoflow** [comment]
when trying to access requirements.txt it tells me: Permission denied
Any ideas why that is?

**Comment by src04c** [comment]
@noahkuhn Have you setup an API key for reddit? Once i setup a script in https://www.reddit.com/prefs/apps and used the api clientID and Keys config_terminal.py. The reddit calls started working fine.

**Comment by noahkuhn** [comment]
I did set one up and added the relevant keys to the config file, but I'm still not getting anything. I found an "API access request" form on Reddit that I filled out wondering if my account needs to be approved first? Not sure. Did you do anything other than create the script on that Reddit prefs/app page?

**Comment by src04c** [comment]
Yeah it does appear to use a Username and Password as well. I tried removing the password from the config and was back to getting null returns.

I did fill out the form for API. But its just a google form and the API keys used in the form I ended up deleting and starting over by creating a new script app. So i dont think its required.. 

So i have 4 items in the config_terminal.py for reddit. 

Client_ID (which is the id that shows up right under the name of the app)
Client secret
Reddit Username
Reddit Password (in cleartext....yuck). I may just create an alt account

Leaving the user_agent at the defaults seemed okay.

**Comment by noahkuhn** [comment]
Did you create it as a "script" application? I think I did "web app" which might be my problem...

**Comment by src04c** [comment]
Yep did it as a script app. 

Used a filler url for the redirect url which was required

The client ID wasn’t obvious either but appears under the name you give it once created.

**Comment by noahkuhn** [comment]
I got this sorted, I was using a .env file and I hadn't prefixed the vars with GT_ thus it wasn't finding them. Duh. Awesome!

**Comment by shadycuz** [comment]
I'm going to close this now. Lots has changed since this was opened. You can run `git checkout main`, `git pull main` to fetch the latest and attempt to reinstall.

The project has been testest to work with Pythons 3.6.8, 3.7 and 3.8. Tensorflow doesn't support python 3.9 yet.

We are using poetry to manage the project and that is one option for installing. The other is to install with the requirements.txt. Feel free to use any virtual environment you want.

Just make sure to run pip install --upgrade pip as that was affecting some people as tensorflow requires pip >= 2.19 (or something like that).

We also have a dockerfile if you would prefer to use docker. We will also soon be offering an "official image" on dockerhub.

Oh and Windows is only supported with docker because pystan is not available for windows (officially).

If you are having problems please open a new issue =).

🚀 GamestonkTerminal 🌔
